### PR TITLE
Fix hostname for containerized omsagent

### PR DESCRIPTION
**Pull request by amitsara**
@Microsoft/omsagent-devs 
@keikhara 

# Issue:
        When omsagent runs inside a container, gethostname returns the hostname of the container (random name) not the actual machine hostname.
        One way to solve this problem is to set the container hostname same as machine name, but this is not possible when host-machine is a private VM inside a cluster.

# Solution:
        Share/mount ‘/etc/hostname’ file with container and omsagent will read hostname from shared file.

### CODE REVIEW

File: source/code/plugins/oms_common.rb [hunk 1]
```diff
@@ -577,6 +577,24 @@ def get_os_version(conf_path = "/etc/opt/microsoft/scx/conf/scx-release")
...
+        # Issue:
+        #   When omsagent runs inside a container, gethostname returns the hostname of the container (random name)
+        #   not the actual machine hostname.
+        #   One way to solve this problem is to set the container hostname same as machine name, but this is not
+        #   possible when host-machine is a private VM inside a cluster.
+        # Solution:
+        #   Share/mount ‘/etc/hostname’ file with container and omsagent will read hostname from shared file.
+
+        begin
+          fileHostName = '/etc/hostname'
```
Comment (robbiezhang) [inline source/code/plugins/oms_common.rb]: please add the unit test

Comment (amitsara) [inline source/code/plugins/oms_common.rb]: Added unit test for this change.

File: test/code/plugins/oms_common_test.rb [hunk 2]
```diff
@@ -172,11 +172,17 @@ def test_get_os_version_missing_field()
 
     def test_get_hostname
       $log = MockLog.new
+      file_hostname = '/etc/hostname'
```
Comment (robbiezhang) [inline test/code/plugins/oms_common_test.rb]: you need to have at least 2 test cases: with the /etc/hostname, and w/o /etc/hostname.

Comment (amitsara) [inline test/code/plugins/oms_common_test.rb]: added tests to verify scenarios when host file exist and when doesn't.
please note that the container file host path is changed to /etc/containerhostname

**Pull request closed without merging**